File: Python/Farmer.py
from Person import Person
from FarmingTool import FarmingTool
from Land import Land

class Farmer(Person):
    def __init__(self, id_person=0, name="", address=""):
        super().__init__(id_person, name, address)
        self.farming_tool_list = []
        self.land_list = []

    # Getter methods
    def get_farming_tool_list(self):
        return self.farming_tool_list

    def get_land_list(self):
        return self.land_list

    # Setter methods
    def set_farming_tool_list(self, daftar):
        self.farming_tool_list = daftar

    def set_land_list(self, daftar):
        self.land_list = daftar

    # Tambah FarmingTool
    def add_farming_tool(self, tool: FarmingTool):
        self.farming_tool_list.append(tool)

    # Tambah Land
    def add_land(self, land: Land):
        self.land_list.append(land)

    # Display Farmer info
    def display_farmer(self, indent=0):
        sp = " " * indent
        print(f"{sp}Farmer ID: {self.get_id_person()}")
        print(f"{sp}Name     : {self.get_name()}")
        print(f"{sp}Address  : {self.get_address()}\n")

        print(f"{sp}    Tools:")
        for tool in self.farming_tool_list:
            tool.display_tools(indent + 8)

        print(f"{sp}    Lands:")
        for land in self.land_list:
            land.display_land(indent + 8)

    def __del__(self):
        pass

File: Java/Farmer.java
import java.util.ArrayList;
import java.util.List;

public class Farmer extends Person {
    private List<FarmingTool> farmingToolList;
    private List<Land> landList;

    // Default constructor
    public Farmer() {
        super(); // panggil constructor Person()
        this.farmingToolList = new ArrayList<>();
        this.landList = new ArrayList<>();
    }

    // Parameterized constructor
    public Farmer(int idPerson, String name, String address) {
        super(idPerson, name, address); // panggil constructor Person
        this.farmingToolList = new ArrayList<>();
        this.landList = new ArrayList<>();
    }

    // Getter methods
    public List<FarmingTool> getFarmingToolList() {
        return farmingToolList;
    }

    public List<Land> getLandList() {
        return landList;
    }

    // Setter methods
    public void setFarmingToolList(List<FarmingTool> daftar) {
        this.farmingToolList = daftar;
    }

    public void setLandList(List<Land> daftar) {
        this.landList = daftar;
    }

    // Tambah FarmingTool
    public void addFarmingTool(FarmingTool tool) {
        this.farmingToolList.add(tool);
    }

    // Tambah Land
    public void addLand(Land land) {
        this.landList.add(land);
    }

    // Display Farmer detail
    public void displayFarmer(int indent) {
        String sp = " ".repeat(indent);

        System.out.println(sp + "Farmer ID: " + getIdPerson());
        System.out.println(sp + "Name     : " + getName());
        System.out.println(sp + "Address  : " + getAddress());
        System.out.println();

        System.out.println(sp + "    Tools:");
        for (FarmingTool tool : farmingToolList) {
            tool.displayTools(indent + 8);
        }

        System.out.println(sp + "    Lands:");
        for (Land land : landList) {
            land.displayLand(indent + 8);
        }
    }
}
